compiler/noirc_frontend/src/hir_def/types.rs

```diff
@@ -1838,19 +1893,36 @@ impl Type {
     /// If this type is a Type::Constant (used in array lengths), or is bound
     /// to a Type::Constant, return the constant as a u32.
     pub fn evaluate_to_u32(&self) -> Option<u32> {
-        if let Some((binding, _kind)) = self.get_inner_type_variable() {
+        self.evaluate_to_field_element(&Kind::u32())
+            .and_then(|field_element| field_element.try_to_u32())
```

Comment:
Do we have any tests which show that we still error for overflowing in intermediate u32 kind computations?
E.g. the equivalent of `U32_MAX + 1 - X` where X > 0

Reply:
No, only tests that show that we're currently erroring in cases where arithmetic generics are `> u32::MAX`. It would be part of https://github.com/noir-lang/noir/issues/6238 so I've added a requirement to test this there.